Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs

```diff
@@ -2096,6 +2186,44 @@ public unsafe void ReadValue(void* buffer, int bufferSize)
             /// <seealso cref="InputAction.ReadValue{TValue}"/>
             /// <seealso cref="ReadValue(void*,int)"/>
             /// <seealso cref="ReadValueAsObject"/>
+            /// <remarks>
+            /// The following example shows how to read the current value of a specific type:
+            /// 
+            /// <example>
+            /// <code>
+            /// using UnityEngine;
+            /// using UnityEngine.InputSystem;
+            ///
+            /// public class Example : MonoBehaviour
+            /// {
+            ///     public InputActionReference move;
+            ///
+            ///     void Awake()
+            ///     {
+            ///         if (move.action != null)
+            ///         {
+            ///             move.action.performed += context =>
+            ///             {
+            ///                 // Note: Assumes the underlying value type is Vector2.
+            ///                 Vector2 value = context.ReadValue&lt;Vector2&gt;();
+            ///                 Debug.Log($"Value is: {value}");
```

Comment:
redundant line in my opinion add something up to a vector (eg the player position)?

Reply:
Fair, I originally had the read value inside the log call but it was more difficult to read. Also I am torn around what to do with the value since the code should compile and ideally do something useful if copied over into a script and entering playmode was my though. Not sure what the suggestion is? To have a public Vector3 direction so its visible in the Inspector?

Reply:
That, or maybe modifiying the ridgid body of the gameobject directly?
eg ridgidBody.velocity = value*speed; (where ridgidbody is a public or serialized field that needs to be assigned in the editor and speed can be a private float field?

Reply:
I would vote for the transform in this case otherwise, if the object has a mesh renderer, otherwise there is nothing to be seen. If rigidBody is used, then we need to add additional detail about that they must reconfigure the system to use FixedUpdate etc or sample value and use additional FixedUpdate to use sampled values. It seems @duckets was fine using Debug.Log at least

Reply:
I see, the only reason why I proposed using the ridgid body was that translating the transform directly is a rather bad practice in general using physics. So I am not sure about how to show case it without messing up with best practices from other areas. Maybe in that case a log would be the safest option.

Reply:
Maybe I am slow but why do we need to use physics?

Reply:
Sorry, I need to rephrase that: for a game using physics translating a player directly using the transform component is a bad idea, since it will not align with the physics of the game (the gravity, wind and other forces). That's why I am wondering if showing that in an example is a bad idea.

Reply:
I am still confused, sorry. I do not see where we use rigid body in examples here? If we were this example would still be valid but only if InputSystem update mode was set to FixedUpdate. If Debug.Log was instead affecting a ridigbody that is. Transform is the way to go when using DynamicUpdate AFAIK, unless sampling in the callback and applying later via FixedUpdate script method.

Reply:
I will mark this resolved know unless someone may help me understand :)